tools/profiler/profiler-process/construct_trace.py

```diff
@@ -251,42 +256,38 @@ def enddefinitions(self, vcd, signals, cur_sig_vals):
             k: v for k, v in control_signal_id_to_names.items() if len(v) > 0
         }
 
-    """
-    Reading through value changes and preserving timestamps to value changes for
-    all signals we "care about".
-    """
-
     def value(self, vcd, time, value, identifier_code, cur_sig_vals):
+        """
+        Reading through value changes and preserving timestamps to value changes for
+        all signals we "care about".
+        """
         int_value = int(value, 2)
         if identifier_code in self.signal_id_to_names:
             signal_names = self.signal_id_to_names[identifier_code]
 
             for signal_name in signal_names:
                 if (
-                    signal_name == f"{self.main_component}.clk" and int_value == 0
-                ):  # ignore falling edges
+                    signal_name == self.clock_name and int_value == 0
+                ):  # ignore falling clock edges
                     continue
-                event = {"signal": signal_name, "value": int_value}
+                event: WaveformEvent = WaveformEvent(signal_name, int_value)
                 if time not in self.timestamps_to_events:
                     self.timestamps_to_events[time] = [event]
                 else:
                     self.timestamps_to_events[time].append(event)
         if identifier_code in self.control_signal_id_to_names:
             signal_names = self.control_signal_id_to_names[identifier_code]
             for signal_name in signal_names:
-                clean_signal_name = (
-                    remove_size_from_name(signal_name)
-                    .split("_go_out")[0]
-                    .replace(self.signal_prefix + ".", "")
-                )
+                clean_signal_name = remove_size_from_name(signal_name).split("_go_out")[
+                    0
+                ]
                 event = {"group": clean_signal_name, "value": int_value}
```

Comment:
This is still using the old dict version, should probably be `WaveformEvent` for
consistency